src/os_net/os_net.c

```diff
@@ -40,64 +40,58 @@ static socklen_t us_l = sizeof(n_us);
 
 
 /* Bind a specific port */
-static int OS_Bindport(u_int16_t _port, unsigned int _proto, const char *_ip, int ipv6)
+int OS_Bindport(char *_port, unsigned int _proto, const char *_ip)
 {
-    int ossock;
-    struct sockaddr_in server;
+    int ossock = 0, s;
+    struct addrinfo hints, *result, *rp;
 
-#ifndef WIN32
-    struct sockaddr_in6 server6;
-#else
-    ipv6 = 0;
-#endif
 
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_family = AF_INET6;    /* Allow IPv4 or IPv6 */
+    hints.ai_protocol = _proto;
     if (_proto == IPPROTO_UDP) {
-        if ((ossock = socket(ipv6 == 1 ? PF_INET6 : PF_INET, SOCK_DGRAM, IPPROTO_UDP)) < 0) {
-            return OS_SOCKTERR;
-        }
+        hints.ai_socktype = SOCK_DGRAM;
     } else if (_proto == IPPROTO_TCP) {
-        int flag = 1;
-        if ((ossock = socket(ipv6 == 1 ? PF_INET6 : PF_INET, SOCK_STREAM, IPPROTO_TCP)) < 0) {
-            return (int)(OS_SOCKTERR);
-        }
-
-        if (setsockopt(ossock, SOL_SOCKET, SO_REUSEADDR,
-                       (char *)&flag,  sizeof(flag)) < 0) {
-            OS_CloseSocket(ossock);
-            return (OS_SOCKTERR);
-        }
+        hints.ai_socktype = SOCK_STREAM;
     } else {
-        return (OS_INVALID);
+        return(OS_INVALID);
     }
+    hints.ai_flags = AI_PASSIVE | AI_ADDRCONFIG | AI_V4MAPPED;
```

Comment:
On Tue, 24 Mar 2015, ddpbsd wrote:

>>      }
>> +    hints.ai_flags = AI_PASSIVE | AI_ADDRCONFIG | AI_V4MAPPED;
>
> It's OpenBSD, so no glibc (older or newer). I don't see AI_V4MAPPED in 
> the compile line, so not sure what's going on there.

Setting up an OpenBSD VM to test...

Antonio Querubin
e-mail:  tony@lavanauts.org
xmpp:  antonioquerubin@gmail.com